docs: backstage setup & plugin integration guide added

anuragkainth opened the pull request:
- Added backstage setup guide using Daytona and devcontainer.
- Added step by step guide to create custom plugins after setting up workspace.
- Added  devcontainer config, screenshots and relevant code snippits.

anuragkainth committed:
Create backstage-guide-v1.md

nibzard commented on docs/backstage-guide-v1.md:
```diff
@@ -0,0 +1,313 @@
+# Getting Started with Backstage: A Developer’s Guide to Setup and Building Custom Plugins
+
+Backstage is an open-source platform, initially developed by Spotify and now part of CNCF, that serves as a unified developer portal. It centralizes all infrastructure tools, services, and documentation, offering a customizable environment where teams can manage projects, streamline workflows, and boost productivity. 
```
you can also mention: https://backstage.spotify.com/

anuragkainth replied:
sure we'll mention this in next commit.

nibzard commented on docs/backstage-guide-v1.md:
````diff
@@ -0,0 +1,313 @@
+```
+2. **Create Daytona Workspace:**
+```bash
+daytona create https://github.com/daytonaio-experiments/starter-backstage.git -t local
````
--code

anuragkainth replied:
okay, we'll compress the command with --code here.

nibzard commented on docs/backstage-guide-v1.md:
````diff
@@ -0,0 +1,313 @@
+daytona create https://github.com/daytonaio-experiments/starter-backstage.git -t local
+```
+
+3. **Select Preferred IDE:**
````
with above change you can skip next two

anuragkainth replied:
sure, we'll skip these after adding --code

nibzard commented on docs/backstage-guide-v1.md:
```diff
@@ -0,0 +1,313 @@
+Backstage is an open-source platform, initially developed by Spotify and now part of CNCF, that serves as a unified developer portal. It centralizes all infrastructure tools, services, and documentation, offering a customizable environment where teams can manage projects, streamline workflows, and boost productivity. 
+
+Built on React and Node, Backstage allows teams to define its functionality through a flexible plugin system, making it a robust framework for creating tailored end-to-end development environments.
+
```
what is the purpose of this guide

anuragkainth replied:
We will ensure that purpose is reflected in the guide.

nibzard commented on docs/backstage-guide-v1.md:
```diff
@@ -0,0 +1,313 @@
+Once you have an organization, proceed to create a new OAuth App:
+
+1. Navigate to [GitHub OAuth Apps](https://github.com/settings/applications/new) and create a new OAuth App for your GitHub user.
+2. For organization-wide integration, go to `https://github.com/organizations/<Organization_Name>/settings/applications`.
```
seems that this is not optional

anuragkainth replied:
This will be necessary if we want members of our github organization to collab with us using backstage.

nibzard commented on docs/backstage-guide-v1.md:
```diff
@@ -0,0 +1,313 @@
+
+5. Use the `HOME > CREATE > REGISTER EXISTING COMPONENT` button to register your service component using the link to the `catalog-info.yaml` file.
+   
+![backstage-doc-2](https://github.com/user-attachments/assets/8a36ca0a-f2b4-4626-a6ed-8ab754f1bb66)
```
I can't see any of the images

anuragkainth replied:
sorry to hear that, i will make sure to add images in a new folder of repo and refer them here.

nibzard commented on docs/backstage-guide-v1.md:
```diff
@@ -0,0 +1,313 @@
+- **postCreateCommand**: Runs `yarn install` to automatically install project dependencies when the container is created.
+- **remoteUser**: Sets "daytona" as the remote user for running commands within the container.
+
+## Building and Customizing Plugins in Backstage
```
this should be mentioned in the intro, as all of the setup is leading to this

anuragkainth replied:
sure, will ensure these changes are reflected in next commit.

nibzard reviewed (changes requested):
please make the changes thanks

anuragkainth committed:
feat: doc image assets added

anuragkainth committed:
rename backstage-image-1.png

anuragkainth committed:
rename backstage-image-2.png

anuragkainth committed:
backstage-image-3.png

anuragkainth committed:
updated backstage-guide-v1.md

anuragkainth commented:
- updated intro of the writeup with changes discussed.
- compressed daytona command with `--code`.
- fixed images. added new assets folder for images.

Please mention in case any more updates are needed. thankyou.

anuragkainth committed:
docs: added link to reference repo inside guide
